crates/eframe/src/native/run.rs

```diff
@@ -233,19 +264,36 @@ fn run_and_return(
             }
         }
 
-        *control_flow = if next_repaint_time <= Instant::now() {
-            if let Some(window) = winit_app.window() {
-                log::trace!("request_redraw");
-                window.request_redraw();
+        // This is for not duplicating redraw requests
+
+        use winit::event::Event;
+        if matches!(
+            event,
+            Event::RedrawEventsCleared | Event::RedrawRequested(_) | Event::Resumed
+        ) {
+            for (window_id, repaint_time) in &windows_next_repaint_times.clone() {
+                if *repaint_time <= Instant::now() {
+                    if let Some(window) = winit_app.window(*window_id) {
+                        log::trace!("request_redraw");
+                        window.borrow().request_redraw();
+                    } else {
+                        windows_next_repaint_times.remove(window_id);
+                    }
+                    control_flow.set_poll();
+                } else {
+                    next_repaint_time = Some(
+                        next_repaint_time.map_or(*repaint_time, |last| last.min(*repaint_time)),
+                    );
+                }
             }
-            next_repaint_time = extremely_far_future();
-            ControlFlow::Poll
-        } else {
+        }
+
+        if let Some(next_repaint_time) = next_repaint_time {
```

Comment:
Something is wrong with this logic: `next_repaint_time` is only ever lowered (`last.min(*repaint_time)`) and never reset. What is it used for? Why do we need it?

Can't we just take the minimum of all `windows_next_repaint_times` instead? That's _almost_ what the code does, but `next_repaint_time` should be decalred  inside the event loop, instead of outside of it

Reply:
> What is it used for? Why do we need it?
This is an assumption, Is used to make the event_loop to wait until is needed and then request redraws!
I was there before, so i adapt it to work with multiples viewports.

> Can't we just take the minimum of all windows_next_repaint_times instead?
Yes, that was what that was supposed to do!